Core/Layer/Worlds/WorldLayer.Render.cs

```diff
@@ -29,7 +32,30 @@ public void RenderWorld(IRenderableSurfaceContext ctx)
 
         SetWorldContextVars();
 
-        ctx.World(m_worldContext, m_renderWorldAction);
+        StatusBarLayoutDef? activeSbar = GetActiveStatusBarLayout();
+        int sbarHeight = activeSbar != null
+            ? (activeSbar.FullscreenRender ? 0 : activeSbar.Height)
+            : (m_config.Hud.StatusBarSize.Value == StatusBarSizeType.Full ? 32 : 0);
+
+        int nativeWidth = ctx.Surface.Dimension.Width;
+        int nativeHeight = ctx.Surface.Dimension.Height;
+
+        int nativeBarHeight = (int)(nativeHeight / 200.0 * sbarHeight);
+
+        int viewportBottom = nativeBarHeight;
+        int viewportHeight = nativeHeight - viewportBottom;
+
+        Box2I viewportArea = new((0, viewportBottom), (nativeWidth, nativeHeight));
+
+        m_worldContext.Viewport = (nativeWidth, viewportHeight);
+
+        ctx.Viewport(viewportArea, () =>
```

Comment:
This lambda needs to be extracted and the Action allocated in the constructor. should be the same as m_renderWorldAction. Otherwise this function gets allocated as a new Action every render frame.